File: genefunk-bootstrap/src/main/java/de/hipp/pnp/genefunk/GeneFunkClassBootstrap.java
package de.hipp.pnp.genefunk;

import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Slf4j
@Component
class GeneFunkClassBootstrap {

	private final GeneFunkClassRepository repository;

	GeneFunkClassBootstrap(GeneFunkClassRepository repository) {
		this.repository = repository;
		initialize();
	}

	protected void initialize() {
		if (this.repository.findByName("Biohacker") == null) {
			this.repository.save(this.initiateBiohacker());
		}
		if (this.repository.findByName("Gunfighter") == null) {
			this.repository.save(this.initiateGunfighter());
		}
	}


	private GeneFunkClass initiateBiohacker() {
		GeneFunkClass biohacker = new GeneFunkClass();
		biohacker.setName("Biohacker");
		return biohacker;
	}

	private GeneFunkClass initiateGunfighter() {
		GeneFunkClass biohacker = new GeneFunkClass();
		biohacker.setName("Gunfighter");
		return biohacker;
	}
}

File: genefunk-bootstrap/src/main/java/de/hipp/pnp/genefunk/GeneFunkClassBootstrap.kt
package de.hipp.pnp.genefunk

import org.springframework.stereotype.Component

@Component
internal class GeneFunkClassBootstrap(private val repository: GeneFunkClassRepository) {
    init {
        initialize()
    }

    protected fun initialize() {
        if (repository.findByName("Biohacker").isEmpty) {
            repository.save(this.initiateBiohacker())
        }
        if (repository.findByName("Gunfighter").isEmpty) {
            repository.save(this.initiateGunfighter())
        }
    }


    private fun initiateBiohacker(): GeneFunkClass {
        val biohacker = GeneFunkClass()
        biohacker.setName("Biohacker")
        return biohacker
    }

    private fun initiateGunfighter(): GeneFunkClass {
        val biohacker = GeneFunkClass()
        biohacker.setName("Gunfighter")
        return biohacker
    }
}
